synapse/server_notices/server_notices_manager.py

```diff
@@ -275,3 +314,51 @@ async def _update_notice_user_profile_if_changed(
                 ratelimit=False,
                 content={"displayname": display_name, "avatar_url": avatar_url},
             )
+
+    async def _update_room_info(
+        self,
+        requester: Requester,
+        room_id: str,
+        info_event_type: str,
+        info_content_key: str,
+        info_value: Optional[str],
+        if_changed: Optional[bool] = True,
+    ) -> None:
+        """
+        Updates a specific notice room's info if it's different from what is set.
+
+        Args:
+            requester: The user who is performing the update.
+            room_id: The ID of the server notice room
+            info_event_type: The event type holding the specific info
+            info_content_key: The key containing the specific info in the event's content
+            info_value: The expected value for the specific info
+            if_changed: Try to read the info first and does not send the event if it's already the same
+        """
+        if if_changed:
+            room_info_event = await self._message_handler.get_room_data(
+                requester,
+                room_id,
+                info_event_type,
+                "",
+            )
+
+            if room_info_event and room_info_event.get(info_content_key) == info_value:
+                return
+
+        if info_value is None:
+            info_value = ""
```

Comment:
Should this be setting `content` to an empty dictionary instead (to delete the value)?

Reply:
I tried en empty `content` but `m.room.topic` is not happy then:
`"'topic' not in content"`

Spec is unclear about that, unlike for `m.room.name` where it explicitly says
> A room with an `m.room.name` event with an absent, null, or empty `name` field should be treated the same as a room with no `m.room.name` event.

related: https://github.com/matrix-org/matrix-spec/commit/044a78730eac66a2b55ed1cf09dd32da80727b13 https://github.com/matrix-org/matrix-spec/issues/1632

We may want to clear the spec and fix synapse to be happy with an empty content.

I can at least do the synapse part if we want to go there.